lib/pliny/middleware/rescue_errors.rb

```diff
@@ -6,14 +6,20 @@ def initialize(app, options = {})
     end
 
     def call(env)
+      error = nil
       @app.call(env)
     rescue Pliny::Errors::Error => e
+      error = e
       Pliny::Errors::Error.render(e)
     rescue => e
+      error = e
       raise if @raise
 
       Pliny::ErrorReporters.notify(e, rack_env: env)
       Pliny::Errors::Error.render(Pliny::Errors::InternalServerError.new)
+    ensure
+      # Leave this in env for CanonicalLogLine.
+      env["pliny.error"] = e
```

Comment:
Can't we just duplicate this line in each `rescue` block? I'm assuming you're explicitly setting it in this way to illustrate the point that it should always be set. If you'd like to retain that explicitness, how about using an `attr_reader :error`? The `error = nil` in the main body weirds me out a little.